src/lib/transition_frontier/full_catchup_tree.ml

```diff
@@ -116,7 +116,7 @@ end
    ancestor is at most k *)
 type t =
   { nodes: Node.t State_hash.Table.t
-  ; states: int Node.State.Enum.Table.t
+  ; states: (int * State_hash.Set.t) Node.State.Enum.Table.t
```

Comment:
Nit: it's good practice to add a record to identify the fields if the type isn't self-documenting

Reply:
After thinking about it twice, I think I just need the set of state hashes there. So I would change the type to be:
```
; states: State_hash.Set.t Node.State.Enum.Table.t
```
This is much simpler.